conans/client/conan_api.py

```diff
@@ -68,7 +68,23 @@
 default_manifest_folder = '.conan_manifests'
 
 
+def api_method_hook(f):
+    def wrapper(api, *args, **kwargs):
+        hook_manager = api.app.hook_manager
+        hook_manager.execute("pre_command", command=f.__name__)
```

Comment:
The name is a bit speculative. If we have a "pre_command" hook is to run something before any command. For other things we have the "pre_build" or "pre_xxx" hooks. I wouldn't like this to evolve towards very complex hooks that do different things before different commands.


I would like to get the feedback from others too.